src/main/java/com/book/mark/service/impl/CreateBookmarksServiceImpl.java

```diff
@@ -367,6 +369,23 @@ public ResponseEntity getResultList(@RequestBody DataProcessingReq dataRequestPa
 				PDDocument doc = PDDocument.load(file1);
 				PDFMerger.addSource(file1);
 				doc.close();
+				
+				if(todoItem != null && todoItem.equalsIgnoreCase(bookMarkPdf)) {
+							readPayloadGenPdfBookmark(dataRequestPayload);
+						}else if(todoItem != null && todoItem.equalsIgnoreCase(generateCSV)){
+							readPayloadGenCSV(dataRequestPayload);
+						}else if(todoItem != null && todoItem.equalsIgnoreCase(genPdfgenCsv)){
+							readPayloadGenPdfBookmark(dataRequestPayload);
+							readPayloadGenCSV(dataRequestPayload);
```

Comment:
Is this line also generating the csv whenever the PDF is bookmarked?

Reply:
@rcoxGSA this story is above validating for all parameters